aten/src/ATen/functorch/ADInterpreters.cpp

```diff
@@ -26,19 +26,22 @@ static void checkForInvalidMutationOnCaptures(
 }
 
 static Tensor materializeGradWrappers(const Tensor& tensor, int64_t current_level) {
+  // we will only mark something as an alias of an unwrapped tensor if it's during user code in a transform.
+  // This state is saved by the during functorch transform context manager
+  const bool mark_as_alias_of_unwrapped = getDuringFunctorchTransform();
```

Comment:
Why is the context manager necessary? Is it a performance optimization or is it for correctness or something else?

Reply:
It's for correctness--during the setup for the transforms (particularly I think like building the dual in jvp), we end up calling some operations that alias the unwrapped input, but it's part of the code to build the wrapper. I'll clarify the comment

Reply:
Off of the top of your head, do you remember which tests failed without this? I'm trying to get a repro to dig in more

Reply:
Somehow I thought it was `vmapvjpvjp` or `jvpvjp` but now I'm not seeing it locally--I have some small naming/comment local changes anyway so I'll just removed this context manager and change and we can see if anything breaks in CI

Reply:
but basically it might have gotten caught up with a couple of other issues, sorry if you've already started investigating!

Reply:
Nah, I just ran the tests locally and see if anything would fail. The tests finished running and everything seems to pass